main.go

```diff
@@ -121,6 +130,50 @@ func main() {
 					os.Exit(1)
 				}
 			}
+		} else if module, _ := tfconfig.LoadModule(dir); len(module.RequiredCore) >= 1 && len(args) == 0 { //if there is a .tfswitchrc file, and no commmand line arguments
+			tfversion := ""
+
+			// we skip duplicated definitions and use only first one
+			tfconstraint := module.RequiredCore[0]
+			tflist, _ := lib.GetTFList(hashiURL, true)
+			fmt.Printf("Reading required version from terraform code, constraint: %s\n", tfconstraint)
+
+			c, err := semver.NewConstraint(tfconstraint)
+			if err != nil {
+				fmt.Println("Error parsing constraint:", err)
+				os.Exit(1)
+			}
+			vs := make([]*semver.Version, len(tflist))
+			for i, r := range tflist {
+				v, err := semver.NewVersion(r)
+				if err != nil {
+					fmt.Printf("Error parsing version: %s", err)
+					os.Exit(1)
```

Comment:
Please correct me if I am wrong but, in my opinion, skipping version in wrong format with proper warning log would be more desirable by an user. If you choose so, please remember to create a 0 length `vs` slice and append to it.

Reply:
What do you mean by this?  
Do you mean we do not exit 1 when there's a wrong format?